### PR TITLE
[MM-843]: Added server testcase for kvstore/plugin_store.go file

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
Added server testcase for kvstore/plugin_store.go file

### CODE REVIEW

File: calendar/utils/kvstore/plugin_store_test.go [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+type MockPluginAPI struct {
+	plugin.API
+	mock.Mock
+}
+
+func (m *MockPluginAPI) KVGet(key string) ([]byte, *model.AppError) {
```
Comment (raghavaggarwal2308) [inline calendar/utils/kvstore/plugin_store_test.go:18]: What was the need for mocking these?

File: go.mod [hunk 2]
```diff
@@ -48,6 +48,7 @@ require (
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/segmentio/backo-go v1.0.1 // indirect
 	github.com/sirupsen/logrus v1.9.3 // indirect
+	github.com/stretchr/objx v0.5.1 // indirect
```
Comment (raghavaggarwal2308) [inline go.mod:51]: Why these changes?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (Kshitij-Katiyar) [reply go.mod:51]: this might be imported by the `"github.com/stretchr/testify/mock"` package used in the testcase

`type MockPluginAPI struct {
	plugin.API
	mock.Mock
}`

File: calendar/utils/kvstore/plugin_store_test.go (hunk 1 above)
Comment (Kshitij-Katiyar) [reply calendar/utils/kvstore/plugin_store_test.go:18]: To implement the PluginAPI. Let me know if we have any better option for that

Comment (raghavaggarwal2308) [reply calendar/utils/kvstore/plugin_store_test.go:18]: But we already have a method for that by MM right? i.e. plugintest

Comment (Kshitij-Katiyar) [reply calendar/utils/kvstore/plugin_store_test.go:18]: pluginTest seems to work for fine type `pluginAPI`, but the API in this file is
`type pluginStore struct {
	api        plugin.API
	ttlSeconds int64
}`

**Review by raghavaggarwal2308 — APPROVED** [review]
